src/frontend/src/meta_client.rs

```diff
@@ -239,4 +248,14 @@ impl FrontendMetaClient for FrontendMetaClientImpl {
     async fn list_hummock_meta_configs(&self) -> Result<HashMap<String, String>> {
         self.0.list_hummock_meta_config().await
     }
+
+    async fn list_compaction_status(
+        &self,
+    ) -> Result<(
+        Vec<CompactStatus>,
+        Vec<CompactTaskAssignment>,
+        Vec<CompactTaskProgress>,
+    )> {
+        self.0.risectl_list_compaction_status().await
+    }
```

Comment:
It is strange to let `list_compaction_status` to return `CompactorStatus`, `CompactTaskAssignment` and `CompactTaskProgress` but the caller actually ignore `CompactTaskAssignment` and `CompactTaskProgress`. I suggest we implement different APIs/RPCs to list different information unless we decide the use one system table to show them all, which also sounds good.

Reply:
thanks, I've refactored it in a new commit.

Reply:
just FYI, view is also supported in rw_catalog.
https://github.com/risingwavelabs/risingwave/blob/483347725eacbe39a9a60b0817ac145aaf205f80/src/frontend/src/catalog/system_catalog/rw_catalog/rw_relations.rs#L24